charts/openshift-metering/templates/hive/hive-scripts-configmap.yaml

```diff
@@ -109,4 +109,32 @@ data:
     export HIVE_METASTORE_HADOOP_OPTS=" -Dhive.log.level=${HIVE_LOGLEVEL} "
     export HIVE_OPTS="${HIVE_OPTS} --hiveconf hive.root.logger=${HIVE_LOGLEVEL},console "
 
+    # Note(tflannag): Hive supports derby (default), mysql, postgres, and oracle databases,
+    # but we only allow users to configure a mysql or postgres database instance.
+    #
+    # The following attempts to determine the database that was configured by the metering
+    # operator by parsing the ConnectionURL parameter in hive-site.xml. We can make the same
+    # assumption that this file exists and that parameter has been set as we control that
+    # file as part of Hive's reconciliation process. We use faq to pull out that key's
+    # value and parse the URL's scheme. We can then use the value of that scheme in order
+    # to avoid determing what database needs to be passed to `schematool -dbType`.
+    #
+    # TODO(tflannag): we may want to use a more dedicated tool in order to parse a URL
+    # as there's currently no validation in place such that invalid ConnectionURL values
+    # being configured may result in unexpected values being returned. Going forward, we
+    # should likely use Ansible to parse this URL for us and setting that environment variable
+    # to avoid determining it using bash.
+    export SCHEMATOOL_BIN="${HIVE_HOME}/bin/schematool"
+    export DATABASE_TYPE=${DATABASE_TYPE:=derby}
+    faq -f xml -o json \
+        '.configuration.property[] | select (.name=="javax.jdo.option.ConnectionURL").value' \
+        "${HIVE_HOME}/conf/hive-site.xml" | \
+        awk -F [/:] '{ print $2 }'
+
+    # Check if there's existing metastore schema for that ${DATABASE_TYPE} and if no
+    # schema exists, i.e. no errors was returned when using the `-info` parameter, attempt
+    # to initialize that schema.
+    echo "Deploying Hive using ${DATABASE_TYPE} as the underlying database"
+    "${SCHEMATOOL_BIN}" -dbType ${DATABASE_TYPE} -info || "${SCHEMATOOL_BIN}" -dbType ${DATABASE_TYPE} -initSchema
```

Comment:
Just a quick comment on this:
- Likely want to move this logic to Ansible as that has URL parsing filters available to us.
- Likely need to expose that `$DATABASE_TYPE` as an environment variable in the StatefuleSet helm chart. Ansible could then set this after parsing the URL in a task.
- In order for this to work with mysql 8.0, we need to update the mysql java connector version in the Hive image.
- Need to use the `schematool -dbType mysql -initSchema` when working with a newer mysql jdbc driver version (one that supports both 5.6/5.7 and 8.0) as there's some compatibility errors that populate when attempting to create the initial metastore tables. I haven't tested whether this is the case for postgres (or derby) - we'll see in the next CI baring any obvious mistakes in the Go code.
- When running `schematool -dbType mysql -initSchema` and that metastore schema already exists in mysql, an error will be produced, which is why we first check the output using the `-info` parameter and if that doesn't fail, then we attempt to initialize the schema.

Reply:
PR for updating the mysql connector java version: https://github.com/kube-reporting/hive/pull/58

Right now, we don't need to wait for that PR to merge, but it also allows us to run tests against mysql 8.0 (and clear out three-related BZs).